### DISCUSSION TITLE
Local vs Central Installation

### DISCUSSION THREAD

**Discussion by gsal** [krvajal/vscode-fortran-support#1048] [General]
Say, I am doing a fresh install of VSCodium, cpptools, fortls, and modern fortran. 

I am actually manually installing the additional extensions right into VSCodium/resource/app/extension directory.

When I install everything in my home directory, things work;  when I install everything in some central location where the user will not have WRITE permission, things do not work.  

Is there something about not having WRITE permissions to the same directory where fortls is installed that prevents it from working?

**Comment by gnikit** [comment]
I think the vscode extension generates some files, to search/not search for the release notes of the extension and to store the lint result artifacts that require write permissions. 

I suspect probably the former is the real issue. 

On a side note, I thought our extension could not be installed in vscodium due to its dependance on cpptools. Is that not the case anymore?

**Comment by gsal** [comment]
Unfortunately, I am kind of bad at taking notes when I am trying to figure how to install something, by the time a setup finally works, I am so confused that I don't know which permutation of the steps worked :-D    This kind of happened to me with vscode and its extensions. 

Initially, I was getting a message about modern fortran not working because its dependency on some unknown cpptools extension; basically, it was not recognizing the cpptools extension.  I tried a couple of different versions (the last one and the one before); also, because my Linux file system seemed to be quoting the cpptools *.vsix  filename due to it containing the '@' character, I had gone and renamed the file after download and previous to installation...and, so, I am under the believe that this was messing up the installation; but that just might be crazy talk. So, I just stopped messing with the filename. 

So, at this point, I have tested time and time again and, when the full installation is done in my home directory, things work; when the full installation is done centrally somewhere where I don't have WRITE permissions, things do not work.  (In case you are wondering how I install where I don't have write permissions, it is because I have two different user id).

Again, by "full installation" I mean instead of installing additional/non-default extensions in the user's ~/.vscode-oss directory, I install them right into the vscode installation itself:  
```
codium --extensions-dir <vscode_home>/resources/app/extensions --install-extension  <ext>.vsix
```

This is unexpected behavior; users use executables from centrally located spots where they do not have WRITE permissions and things work just fine; if a particular program needs to write somewhere, it either does it in /tmp or figures out the users home directory.

**Reply by gnikit** [reply]
I think you missed this

> I think the vscode extension generates some files, to search/not search for the release notes of the extension and to store the lint result artifacts that require write permissions.

Specifically the extension caches the result at its install location https://github.com/fortran-lang/vscode-fortran-support/blob/0e65a7bca9a653d37c3c3327f5fc350f2a44350d/src/extension.ts#L129
If you have a better way of doing this that is cross-platform and does not require a lot maintainance feel free to open a PR. Might want to verify that this is the problem.

**Reply by gnikit** [reply]
I still don't know how you got the vscode cpptools to work in vscodium. The extension should not be able to install without them.

**Comment by gsal** [comment]
```
 fs.writeFileSync(path.join(__dirname, 'displayReleaseNotes.txt'), 'false'); 
```
Is that TypeScript? I am not a web developer, I am a Fortran programmer.  

What I am saying is that it is bad practice and altogether incorrect to have a program write to the location where it itself is installed. At least in Linux where the concept of permissions exists and where programs are installed by root, when a user is running a program, they will not have WRITE permission to the location where the program is installed.

Other than that, I am sure TypeScript or JavaScript have a way to find out the user's home directory; from there, this fortran-support extension could go into .vscode and write in there. 

```
I still don't know how you got the vscode cpptools to work in vscodium. The extension should not be able to install without them
```
I don't quite understand this statement; maybe, in all my rambling, I say something not clear.   I did install cpptools.
